src/Bicep.Cli.IntegrationTests/UseRecentModuleVersionsIntegrationTests.cs

```diff
@@ -0,0 +1,822 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.IO.Abstractions;
+using System.IO.Abstractions.TestingHelpers;
+using Azure;
+using Azure.Containers.ContainerRegistry;
+using Azure.Identity;
+using Bicep.Cli.UnitTests;
+using Bicep.Cli.UnitTests.Assertions;
+using Bicep.Core.Analyzers.Linter.Rules;
+using Bicep.Core.Configuration;
+using Bicep.Core.Modules;
+using Bicep.Core.Registry;
+using Bicep.Core.Registry.Oci;
+using Bicep.Core.Registry.PublicRegistry;
+using Bicep.Core.Samples;
+using Bicep.Core.UnitTests;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Baselines;
+using Bicep.Core.UnitTests.Mock;
+using Bicep.Core.UnitTests.Registry;
+using Bicep.Core.UnitTests.Utils;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Bicep.Cli.IntegrationTests;
+
+[TestClass]
+public class UseRecentModuleVersionsIntegrationTests : TestBase
+{
+    public static BicepModuleMetadata[] DefaultModulesMetadata = [
+        new BicepModuleMetadata(
+                    "fake/avm/res/app/container-app",
+                    ["0.2.0"],
+                    new Dictionary<string, BicepModuleTagPropertiesEntry>().ToImmutableDictionary()),
```

Comment:
```suggestion
                    ImmutableDictionary<string, BicepModuleTagPropertiesEntry>.Empty),
```

or `[]` should just work?